umap/static/umap/js/modules/rules.js

```diff
@@ -0,0 +1,240 @@
+import { DomUtil, DomEvent, stamp } from '../../vendors/leaflet/leaflet-src.esm.js'
+import * as Utils from './utils.js'
+import { translate } from './i18n.js'
+
+class Rule {
+  constructor(map, condition = '', options = {}) {
+    this.map = map
```

Comment:
I'm wondering if it would it be possible to not pass the `map` object here? You would need to find ways to handle `isDirty` and the `render` call in a different way, maybe with events?

Reply:
This is a pattern we already have elsewhere in uMap, including in modules. Also I remember removing events call from the `requests` module (done by inheriting Leaflet `Class`), so I'm not exactly sure what is the pattern we'd encourage.

Anyway, I'm ok to challenge this pattern, but I'd prefer to do it in a separate PR and not to charge this one.

Reply:
The goal would be to move away from having this global `map` superobject. I'm okay to follow the discussion elsewhere.